internal/cloud/awscloud/secure-instance.go

```diff
@@ -228,6 +228,9 @@ func (a *AWS) TerminateSecureInstance(si *SecureInstance) error {
 }
 
 func (a *AWS) terminatePreviousSI(hostInstanceID string) (string, error) {
+	var retryCount int
+	const maxRetries = 3
```

Comment:
This number will need to be way higher actually, like idk how long each try takes but I think we should wait more than 3 times. Like up to 10 minutes or so.

Reply:
I'm not even sure if we should do it or if we should let the waiter do it?

looking at https://github.com/aws/aws-sdk-go-v2/blob/service/ec2/v1.191.0/service/ec2/api_op_DescribeInstances.go#L1383 ; the thing that calls the retryable takes care of everything there.

Reply:
we can also increase the `MinDelay`
so `MinDelay = 60 * time.Second` and `maxRetries = 10` ?

Reply:
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-lifecycle.html I think we should basically do the same except allow "stopping", since I think we might ebs-backed instances sometimes? Not sure. This honestly seems like a bug in the sdk.

Reply:
Hmm, let's maybe try without the custom retryfunction but just increase the mindelay and maxretries?

Reply:
as in theory it should wait for 1 hour (as given by the `maxWaitDur` of `.Wait()`) we probably need to retry on "Failure" (what ever that is)